allensdk/brain_observatory/behavior/behavior_project_cache/behavior_ecephys_project_cache.py

```diff
@@ -0,0 +1,283 @@
+        get_ecephys_session_table()
+        get_behavior_session_table()
+        get_probe_table()
+        get_channel_table()
+        get_unit_table
+
+        Provides methods for instantiating session objects
+        from the nwb files:
+        get_ecephys_session() to load BehaviorEcephysSession
+        get_behavior_sesion() to load BehaviorSession
+
+        Provides tools for downloading data:
+
+        Will download data from the s3 bucket if session nwb file is not
+        in the local cache, othwerwise will use file from the cache.
+
+        Parameters
+        ==========
+        fetch_api :
+            Used to pull data from remote sources, after which it is locally
+            cached.
+        fetch_tries :
+            Maximum number of times to attempt a download before giving up and
+            raising an exception. Note that this is total tries, not retries.
+            Default=2.
+        """
+
+        self.fetch_api = fetch_api
+        self.cache = None
+
+        self.fetch_tries = fetch_tries
+        self.logger = logging.getLogger(self.__class__.__name__)
+
+    @property
+    def manifest(self):
+        if self.cache is None:
+            api_name = type(self.fetch_api).__name__
+            raise NotImplementedError(f"A {type(self).__name__} "
+                                      f"based on {api_name} "
+                                      "does not have an accessible manifest "
+                                      "property")
+        return self.cache.manifest
+
+    @classmethod
+    def from_s3_cache(cls, cache_dir: Union[str, Path],
+                      bucket_name: str = "visual-behavior-ecephys-data",
+                      project_name: str = "visual-behavior-ecephys"
+                      ) -> "VisualBehaviorEcephysProjectCache":
+        """instantiates this object with a connection to an s3 bucket and/or
+        a local cache related to that bucket.
+
+        Parameters
+        ----------
+        cache_dir: str or pathlib.Path
+            Path to the directory where data will be stored on the local system
+
+        bucket_name: str
+            for example, if bucket URI is 's3://mybucket' this value should be
+            'mybucket'
+
+        project_name: str
+            the name of the project this cache is supposed to access. This
+            project name is the first part of the prefix of the release data
+            objects. I.e. s3://<bucket_name>/<project_name>/<object tree>
+
+        Returns
+        -------
+        VisualBehaviorEcephysProjectCache instance
+
+        """
+        fetch_api = BehaviorEcephysProjectCloudApi.from_s3_cache(
+                cache_dir, bucket_name, project_name,
+                ui_class_name=cls.__name__)
+
+        return cls(fetch_api=fetch_api)
+
+    @classmethod
+    def from_local_cache(
+        cls,
+        cache_dir: Union[str, Path],
+        project_name: str = "visual-behavior-ecephys",
```

Comment:
Same comment above: should we even be allowing users to change project_name?

Reply:
now it is a class variable.